src/object/srv_obj.c

```diff
@@ -86,14 +86,91 @@ obj_gen_dtx_mbs(struct daos_shard_tgt *tgts, uint32_t *tgt_cnt,
 		mbs->dm_tgt_cnt = j;
 		mbs->dm_grp_cnt = 1;
 		mbs->dm_data_size = size;
-		mbs->dm_flags = DMF_MODIFY_SRDG;
+		if (!is_ec)
+			mbs->dm_flags = DMF_SRDG_REP;
 	}
 
 	*p_mbs = mbs;
 
 	return 0;
 }
 
+static int
+obj_handle_uncertainty(struct dtx_handle *dth, struct ds_cont_child *cont)
+{
+	struct dtx_share_peer	*dsp;
+	struct dtx_share_peer	*tmp;
+	int			 rc = 0;
+
+	d_list_for_each_entry_safe(dsp, tmp, &dth->dth_share_tbd_list,
+				   dsp_link) {
+		if (dsp->dsp_leader == PO_COMP_ID_ALL) {
+			rc = ds_pool_elect_dtx_leader(cont->sc_pool->spc_pool,
+						      &dth->dth_leader_oid,
+						      dsp->dsp_ver);
+			if (rc < 0) {
+				D_ERROR("Failed to find DTX leader for "DF_DTI
+					": "DF_RC"\n",
+					DP_DTI(&dsp->dsp_xid), DP_RC(rc));
+				return rc;
+			}
+
+			dsp->dsp_leader = rc;
+		}
+
+		d_list_del(&dsp->dsp_link);
+
+		rc = dtx_probe(cont, &dsp->dsp_xid, dsp->dsp_leader);
+		switch (rc) {
+		case DTX_ST_PREPARED:
+			/* Not committable yet. */
+			d_list_add_tail(&dsp->dsp_link,
+					&dth->dth_share_act_list);
+			break;
+		case DTX_ST_COMMITTABLE:
+			/* Committable, will be committed soon. */
+			d_list_add_tail(&dsp->dsp_link,
+					&dth->dth_share_cmt_list);
+			break;
+		case DTX_ST_COMMITTED:
+			/* Has been committed on leader, we may miss related
+			 * commit request, so let's commit it locally.
+			 */
+			rc = vos_dtx_commit(cont->sc_hdl, &dsp->dsp_xid,
+					    1, NULL);
+			D_FREE(dsp);
+			if (rc == -DER_NONEXIST)
+				rc = 0;
+			break;
+		case -DER_NONEXIST:
+			/* The leader does not have related DTX info,
+			 * we may miss related DTX abort request, so
+			 * let's abort it locally.
+			 */
+			rc = vos_dtx_abort(cont->sc_hdl, DAOS_EPOCH_MAX,
+					   &dsp->dsp_xid, 1);
+			D_FREE(dsp);
+			if (rc == -DER_NONEXIST)
+				rc = 0;
+		default:
```

Comment:
(style)  Possible switch case/default not preceeded by break or fallthrough comment